widget/scroller.go

```diff
@@ -485,31 +369,58 @@ func (s *ScrollContainer) MinSize() fyne.Size {
 	return s.BaseWidget.MinSize()
 }
 
+// Refresh causes this widget to be redrawn in it's current state
+func (s *ScrollContainer) Refresh() {
+	s.updateOffset(0, 0)
+	s.refreshWithoutOffsetUpdate()
+}
+
+func (s *ScrollContainer) refreshWithoutOffsetUpdate() {
+	s.BaseWidget.Refresh()
+}
+
+// Resize sets a new size for the scroll container.
+func (s *ScrollContainer) Resize(size fyne.Size) {
```

Comment:
I don't think this is needed - BaseWidget does the same

Reply:
Not exactly. `BaseWidget` performs a `Layout` via the renderer.
This would not update the widget's offset.
Also I dislike the idea to call renderer methods from the widget.

Reply:
What if the offset was a ratio rather than an absolute? then layout would do the right thing and this would go away? Not a suggestion, just a question - I imagine that may be a lot of work.

Reply:
I can't see where the change to a ratio would remove the necessity to adjust the position of the content to changes of the container's size.

Reply:
Fair point